src/System.Private.ServiceModel/src/System/ServiceModel/Channels/Connection.cs

```diff
@@ -31,6 +31,62 @@ public interface IConnection
         int EndRead();
     }
 
+    public static class IConnectionExtensions
+    {
+        public static async Task WriteAsync(this IConnection connection, byte[] buffer, int offset, int size, bool immediate, TimeSpan timeout)
```

Comment:
Please add a comment that these methods shouldn't be used on message send/receive and are convenience methods for the open/close code paths. I would actually prefer this class to be in a separate file in a unique namespace so that intellisense doesn't show them unless you have explicitly added a using statement for the namespace. The BeginWrite/BeginRead methods are intended to be zero allocation and I would like to discourage the usage of these helpers on hot code paths.